tools/autograd/derivatives.yaml

```diff
@@ -1151,6 +1151,12 @@
   self: max_unpool3d_backward(grad, self, indices, output_size, stride, padding)
   indices: non_differentiable
 
+- name: convolution_generic(Tensor input, Tensor weight, Tensor? bias, int[] stride, int[] padding, int[] dilation, bool transposed, int[] output_padding, int groups) -> Tensor
+  input, weight, bias: convolution_generic_backward(grad, input, weight, bias, stride, padding, dilation, transposed, output_padding, groups)
+
+- name: convolution_generic_backward(Tensor grad_output, Tensor input, Tensor weight, Tensor? bias, int[] stride, int[] padding, int[] dilation, bool transposed, int[] output_padding, int groups) -> (Tensor grad_input, Tensor grad_weight, Tensor grad_bias)
+  grad_output, input, weight: convolution_generic_double_backward(grads[0], grads[1], grads[2], grad_output, input, weight, stride, padding, dilation, transposed, output_padding, groups)
```

Comment:
I think that you can use the existing double backward function here, I believe it's generic.

Reply:
I didn't use `_convolution_double_backward` mainly because we have `deterministic/cudnn_enabled` backend specific flags and `output_mask` which is an implementation detail in our autograd. (I think `output_mask` is fine, need to double check with XLA ppl). 
We need carefully design the generic function signatures so that it contains all info needed to calculate backward in existing and future backends, so that it works as API we provide. Ideally the less often we change this function in the future, the better :D

Reply:
Hmm, the presence of `deterministic` and `cudnn_enabled` is a wart, I agree. I suppose we can make another version of this function that just reads these out of the global values and let you call that. `output_mask` is important: it tells you what outputs you actually want to compute. It's relevant to XLA too.